pyro/generic/generic.py

```diff
@@ -36,55 +73,83 @@ def pyro_backend(*aliases, **new_backends):
     or by providing a dict mapping module name to backend module name.
     Standard backends include: pyro, minipyro, funsor, and numpy.
     """
+    global _BACKEND
+    old_backend_name = _BACKEND
     if aliases:
```

Comment:
nit: not related to this PR, but I think we can simplify the logic by using the signature: `def pyro_backend(alias=None, **new_backends):`, then `if alias is not None` here.

Reply:
The `*args` pattern avoids any possible masking of `**kwargs`. While it is a little more complex, it safely handles the edge case of a backend providing a library named 'alias'.